### PR TITLE
Add module documentation for the pauli module

### PR DESCRIPTION
**Pull request by Jaybsoni**
Make sure module documentation highlights the new `PauliWord` and `PauliSentence` classes

### CODE REVIEW

**Review by antalszava — APPROVED** [review]
Looks good! :clap: Two comments, approval is conditional on the `PauliWord` rendering.

File: doc/code/qml_pauli.rst [hunk 1]
```diff
@@ -16,71 +16,74 @@ for Pauli-word partitioning functionality used in measurement optimization.
...
+>>> pw1 * pw2
+(Z(0), 1j)
+>>> pw1.to_mat(wire_order=[0,1])
+array([[ 0,  0,  1,  0],
+       [ 0,  0,  0, -1],
+       [ 1,  0,  0,  0],
+       [ 0, -1,  0,  0]])
+
+
+The `PauliSentence` class represents linear combinations of Pauli words. Using a
```
Comment (josh146) [inline doc/code/qml_pauli.rst:57]:
```suggestion
The :class:`~.PauliSentence` class represents linear combinations of Pauli words. Using a
```

File: doc/code/qml_pauli.rst [hunk 2]
```diff
@@ -16,71 +16,74 @@ for Pauli-word partitioning functionality used in measurement optimization.
...
->>> pennylane.pauli import pauli_word_to_matrix
->>> wire_map = {'a' : 0, 'b' : 1}
->>> pauli_word = qml.PauliZ('b')  # corresponds to Pauli 'IZ'
->>> pauli_word_to_matrix(pauli_word, wire_map=wire_map)
-array([[ 1.,  0.,  0.,  0.],
-       [ 0., -1.,  0., -0.],
-       [ 0.,  0.,  1.,  0.],
-       [ 0., -0.,  0., -1.]])
+`PauliWord` is a lightweight class which uses a dictionary approach to represent
+Pauli words. A `PauliWord` can be instantiated by passing a dictionary of wires
```
Comment (josh146) [inline doc/code/qml_pauli.rst:36]:
```suggestion
Pauli words. A ``PauliWord`` can be instantiated by passing a dictionary of wires
```

just a reminder to always use double backticks in rest :)

Comment (Jaybsoni) [reply doc/code/qml_pauli.rst]: Added 💪🏼

**Pull request merged**